### PR TITLE
Use flex for screen

### PR DESCRIPTION
**Pull request by gabriel**
We don't need a fixed height (and height isn't always fixed). It's appropriate to pass flex: 1 for all "main" content.

This fixes issue where status bar height changes would hide part of nav, and now since it can handle variable height, works as expected.

### CODE REVIEW

**Comment by gabriel** [comment]
@keybase/react-hackers

**Comment by gabriel** [comment]
https://keybase.atlassian.net/browse/DESKTOP-3245

**Comment by chrisnojima** [comment]
FYI if you follow the naming convention the team uses (gabriel/DESKTOP-3245-fix-nav-height) jira will auto link this stuff

**Comment by gabriel** [comment]
Bump

**Comment by chrisnojima** [comment]
This has bad side effects. if you type in chat you'll see it pushes the global nav up etc

**Comment by gabriel** [comment]
iOS works well with KeyboardAvoidingView and Android doesn't.

So I split the "main nav stack" into 2 versions.
The iOS is:

```jsx
<Box style={flexOne}>
      <NativeKeyboardAvoidingView>
           Content
      </NativeKeyboardAvoidingView>
      Nav
</Box>
```

Android uses the old approach in master with dimensions height - tab bar height.

File: shared/nav.native.js [hunk 1]
```diff
@@ -82,38 +72,66 @@ function renderMainStackRoute (route) {
   )
 }
 
-function MainNavStack (props: Props) {
+function cardStackShim (props: Props) {
```
Comment (chrisnojima) [inline shared/nav.native.js]: This is effectively a stateless component so it should be capitalized and treated like a component

Comment (chrisnojima) [inline shared/nav.native.js]: 'CardStackShim' which lets you do <CardStackShim />

File: shared/nav.native.js [hunk 2]
```diff
@@ -82,38 +72,66 @@ function renderMainStackRoute (route) {
...
+        renderRoute={renderMainStackRoute}
+        onNavigateBack={props.navigateUp}
+      />
+      {![chatTab].includes(props.routeSelected) && <Offline reachability={props.reachability} appFocused={true} />}
+      <GlobalError />
+    </Box>
+  )
+}
+
+function tabBar (props: Props) {
```
Comment (chrisnojima) [inline shared/nav.native.js]: TabBar

File: shared/nav.native.js [hunk 3]
```diff
@@ -82,38 +72,66 @@ function renderMainStackRoute (route) {
...
+      username={props.username}
+      badgeNumbers={{
+        [chatTab]: props.chatBadge,
+        [folderTab]: props.folderBadge,
+      }}
+    />
+  )
+}
+
+function mainNavStackIOS (props: Props) {
```
Comment (chrisnojima) [inline shared/nav.native.js]: you can just have a IOS and Android wrapper that takes in a child instead of having to instantiate cardStackShim inside both

Comment (chrisnojima) [inline shared/nav.native.js]: ie something like

```
const forIOS = ({shim, tabBar}) => (
    <Box style={flexOne}>
       <NativeKeyboardAvoidingView behavior={'padding'} style={sceneWrapStyleUnder}>
       {shim}
       ...
)

const forAndroid = ({shim, tabBar}) => (
    <Box style={flexOne}>
         <Box style={!props.hideNav ? styleScreenSpaceAndroid : flexOne}>
             {shim}
)


const Container = isAndroid ? forAndroid : forIOS
const shim = <CardStackShim key={...} stack={props....}/>
const tabBar = <TabBar onTabClicked={...} ... />

return <Container shim={shim} tabBar={tabBar} />



```

**Comment by gabriel** [comment]
I think ready

**Pull request merged**